probe/kubernetes/client.go

```diff
@@ -56,14 +58,39 @@ type client struct {
 }
 
 // runReflectorUntil is equivalent to cache.Reflector.RunUntil, but it also logs
-// errors, which cache.Reflector.RunUntil simply ignores
+// errors (which cache.Reflector.RunUntil simply ignores) and backs off exponentially
+// on errors.
 func runReflectorUntil(r *cache.Reflector, resyncPeriod time.Duration, stopCh <-chan struct{}) {
-	loggingListAndWatch := func() {
-		if err := r.ListAndWatch(stopCh); err != nil {
+	select {
+	case <-stopCh:
+		return
+	default:
+	}
+
+	var err error
+	wait := resyncPeriod
+	for {
+		func() {
+			defer runtime.HandleCrash()
```

Comment:
Is this needed?

From the HandleCrash documentation: 

```
// TODO: remove this function. We are switching to a world where it's safe for
// apiserver to panic, since it will be restarted by kubelet.
```